src/core/lib/resource_quota/periodic_update.cc

```diff
@@ -0,0 +1,60 @@
+// Copyright 2022 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <grpc/support/port_platform.h>
+
+#include "src/core/lib/resource_quota/periodic_update.h"
+
+namespace grpc_core {
+
+bool PeriodicUpdate::MaybeEndPeriod() {
+  auto now = ExecCtx::Get()->Now();
+  Duration time_so_far = now - period_start_;
+  if (time_so_far < period_) {
+    // Double the number of updates remaining until the next period.
+    int64_t better_guess;
+    if (time_so_far.millis() == 0) {
+      better_guess = expected_updates_per_period_ * 2;
+    } else {
+      const double scale = period_.seconds() / time_so_far.seconds();
+      if (scale > 2) {
+        better_guess = expected_updates_per_period_ * 2;
+      } else {
```

Comment:
Maybe comment that the scale in this case is still >=1 for clarity?

Reply:
done